conductor/lib/common.py

```diff
@@ -440,14 +444,20 @@ def validate_api_key(config):
         Returns: None
 
         """
-        if 'api_key_path' not in config:
-            config['api_key_path'] = os.path.join(base_dir(), 'auth', 'conductor_api_key')
-        api_key_path = config['api_key_path']
+        # If the key isn't defined, than check for a path with the key
+        if 'api_key' in config:
+            logger.debug("'api_key' is already defined. Ignoring 'api_key_path'")
+            return
+
+        if 'api_key_path' not in config:          
+            api_key_path = Config.get_default_api_key_path()
+            logger.debug("'api_key_path' not found in config, checking base dir ({}) for api key path".format(api_key_path))
 
         #  If the API key doesn't exist, then no biggie, just bail
         if not os.path.exists(api_key_path):
-            # config['api_key'] = None
-            return
+            logger.debug("No API key file found '{}'. Not using.".format(api_key_path))
```

Comment:
Use `warning` log level.

Reply:
This message will be printed even if no config option for the api key path was set. Won't it be confusing for the user if they get a warning in that case?

Reply:
Isn't it a critical error if there is no file where the default api key path is expected to be? This function is named `validate_api_key`, a silent failure isn't what I'd expect.

Reply:
I don't disagree.. but that was the existing logic. I added the log message to help debug when there are issues. Fixing it would require a re-work of the whole flow.